### PR TITLE
Use theme attribute to update MapView map style URL

### LINKED ISSUE #994: Map style didn't change to Dark/Night mode
Noticed this while testing:

night_mode_not_working

Map style didn't change to Dark/Night mode whereas the rest of the UI did. Noting that it happened right at 10pm.

### PR DESCRIPTION
**Pull request by danesfeder**
Closes #994 

@mapbox/maps-android the `mapbox_styleUrl` is taken into account every time the `MapView` is rendered for the first time correct?  

If we set the style URL from `navigationViewMapStyle` there shouldn't be any possible issues with caching and we can remove that method from `ThemeSwitcher`.

FWIW, I created an example that used the cancel btn to change the theme and recreate (simulating if this we to happen mid-drive) and I wasn't able to reproduce.  @Guardiola31337 if you can provide a reliable way to reproduce, that would be great.

### CODE REVIEW

**Comment by LukasPaczos** [comment]
@danesfeder correct, it's taken into account every time the `MapView` is inflated.

**Comment by Guardiola31337** [comment]
@danesfeder 
> FWIW, I created an example that used the cancel btn to change the theme and recreate (simulating if this we to happen mid-drive) and I wasn't able to reproduce.

Could you add the example you wrote to try to reproduce #994 as part of this PR? I think that would be great to have it anyway so we can test different scenarios or any regressions that might appear.

**Comment by Guardiola31337** [comment]
I can confirm that the changes implemented in this PR don't solve #994 👀 

night_style_map_issue

When bringing app into the foreground after backgrounding nothing happened either, the map style didn't change 😓 

Even after rotating the screen 👀 

night_style_map_issue_landscape

My hunch either a regression from #939 or the `MapView` is caching somehow. Any hints @mapbox/maps-android?

**Comment by Guardiola31337** [comment]
@danesfeder latest changes seem to fix the issue of changing the map style automatically between day/night mode (I still have to check if it changes automatically from night to day mode, day to night works 👌) but now during night it doesn't load the proper style at startup time (first time) 👀 

night_mode_startup_issue

Also noticed that if you rotate the screen loads the proper map style (night in the example above) but the upcoming maneuver arrow disapears.

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 1]
```diff
@@ -432,6 +433,15 @@ public void onInstructionListVisibilityChanged(boolean visible) {
     });
   }
 
+  private void updateSavedInstanceStateMapStyle(@NonNull MapView mapView, @Nullable Bundle savedInstanceState) {
+    String mapStyleUrl = ThemeSwitcher.retrieveMapStyle(getContext());
+    if (savedInstanceState != null) {
+      savedInstanceState.putString(MapboxConstants.STATE_STYLE_URL, mapStyleUrl);
+    } else {
+      mapView.setStyleUrl(mapStyleUrl);
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: @Guardiola31337 found the issue - we were previously only updated when the saved instance state _was not_ null, thus it was always day mode style.

**Comment by danesfeder** [comment]
@Guardiola31337 this is ready for 👀

**Comment by Guardiola31337** [comment]
@danesfeder 
> found the issue - we were previously only updated when the saved instance state was not null, thus it was always day mode style.

Nah, that doesn't solve it (at least completely) 👀 

theme_switching_issue

It fails when closing and then reopening. It seems something with initialization

**Comment by danesfeder** [comment]
I was able to reproduce https://github.com/mapbox/mapbox-navigation-android/pull/1018#issuecomment-398740432 and fixed it by setting the style URL again in the XML, but keeping the saved instance state check in `NavigationView`.

Currently looking into a public API to toggle the style (day vs. night) on the fly.  Not sure if it's totally in scope of this PR but would make it easy to create an example.

**Comment by danesfeder** [comment]
@Guardiola31337 @devotaaabel this PR is ready for 👀

**Comment by Guardiola31337** [comment]
It's working now 🎉 

theme_night_mode_working

I'd need to test if the theme changes from night to day but this is definitely good to go 🚀 

Well done @danesfeder 👏

**Pull request merged**